### ISSUE TITLE
panic on bodyclose

### ISSUE THREAD

**Issue by powerman** [golangci/golangci-lint#608]
This happens just once on CI and wasn't reproduce on rebuilding same commit on CI.

```
+ golangci-lint run
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x99471f]

goroutine 4084 [running]:
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).run(0xc0000b8dc0, 0xc01523e140, 0x10, 0xd1d9e0, 0x45f692d19a978201, 0xc034e87250)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:57 +0x1af
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).execOnce(0xc01d5bf4a0)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:382 +0x68a
sync.(*Once).Do(0xc01d5bf4a0, 0xc0015d3f90)
	/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/sync/once.go:44 +0xb3
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).exec(0xc01d5bf4a0)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:303 +0x50
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll.func1(0xc01d5bf4a0)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:291 +0x34
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:297 +0x11b
```

```
$ golangci-lint --version
golangci-lint has version 1.17.1 built from 4ba2155 on 2019-06-10T09:06:49Z

$ go version && go env
go version go1.12.6 linux/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/app/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build343332231=/tmp/go-build -gno-record-gcc-switches"

$ cat .golangci.yml
# This file contains all available configuration options
# with their default values.

# options for analysis running
run:
  # default concurrency is a available CPU number
  # concurrency: 4

  # timeout for analysis, e.g. 30s, 5m, default is 1m
  # deadline: 1m

  # exit code when at least one issue was found, default is 1
  # issues-exit-code: 1

  # include test files or not, default is true
  # tests: true

  # list of build tags, all linters use it. Default is empty list.
  build-tags:
    - integration

  # which dirs to skip: they won't be analyzed;
  # can use regexp here: generated.*, regexp is applied on full path;
  # default value is empty list, but next dirs are always skipped independently
  # from this option's value:
  #   	vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
  # skip-dirs:
  #   - src/external_libs
  #   - autogenerated_by_my_lib

  # which files to skip: they will be analyzed, but issues from them
  # won't be reported. Default value is empty list, but there is
  # no need to include all autogenerated files, we confidently recognize
  # autogenerated files. If it's not please let us know.
  skip-files:
    - "\\.html\\.go$"
    - "\\.json\\.go$"
    - "\\.sql\\.go$"
    - "_reform\\.go$"

  # by default isn't set. If set we pass it to "go list -mod={option}". From "go help modules":
  # If invoked with -mod=readonly, the go command is disallowed from the implicit
  # automatic updating of go.mod described above. Instead, it fails when any changes
  # to go.mod are needed. This setting is most useful to check that go.mod does
  # not need updates, such as in a continuous integration and testing system.
  # If invoked with -mod=vendor, the go command assumes that the vendor
  # directory holds the correct copies of dependencies and ignores
  # the dependency descriptions in go.mod.
  modules-download-mode: vendor


# output configuration options
output:
  # colored-line-number|line-number|json|tab|checkstyle|code-climate, default is "colored-line-number"
  # format: colored-line-number

  # print lines of code with issue, default is true
  # print-issued-lines: true

  # print linter name in the end of issue text, default is true
  # print-linter-name: true


# all available settings of specific linters
linters-settings:
  errcheck:
    # report about not checking of errors in type assetions: `a := b.(MyStruct)`;
    # default is false: such cases aren't reported by default.
    # check-type-assertions: false

    # report about assignment of errors to blank identifier: `num, _ := strconv.Atoi(numStr)`;
    # default is false: such cases aren't reported by default.
    # check-blank: false

    # [deprecated] comma-separated list of pairs of the form pkg:regex
    # the regex is used to ignore names within pkg. (default "fmt:.*").
    # see https://github.com/kisielk/errcheck#the-deprecated-method for details
    # ignore: fmt:.*,io/ioutil:^Read.*

    # path to a file containing a list of functions to exclude from checking
    # see https://github.com/kisielk/errcheck#excluding-functions for details
    # exclude: .errcheck.excludes
  govet:
    # report about shadowed variables
    check-shadowing: true

    # settings per analyzer
    settings:
      printf: # analyzer name, run `go tool vet help` to see all analyzers
        funcs: # run `go tool vet help printf` to see available settings for `printf` analyzer
          # - (github.com/powerman/structlog.Logger).Fatalf
          # - (github.com/powerman/structlog.Logger).Panicf
          # - (github.com/powerman/structlog.Logger).Printf
  golint:
    # minimal confidence for issues, default is 0.8
    # min-confidence: 0.8
  gofmt:
    # simplify code: gofmt with `-s` option, true by default
    # simplify: true
  goimports:
    # put imports beginning with prefix after 3rd-party packages;
    # it's a comma-separated list of prefixes
    # local-prefixes: github.com/powerman
  gocyclo:
    # minimal code complexity to report, 30 by default (but we recommend 10-20)
    min-complexity: 15
  maligned:
    # print struct with more effective memory layout or not, false by default
    suggest-new: true
  dupl:
    # tokens count to trigger issue, 150 by default
    threshold: 100
  goconst:
    # minimal length of string constant, 3 by default
    # min-len: 3
    # minimal occurrences count to trigger, 3 by default
    # min-occurrences: 3
  depguard:
    list-type: blacklist
    include-go-root: true
    packages:
      - log
  misspell:
    # Correct spellings using locale preferences for US or UK.
    # Default is to use a neutral variety of English.
    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
    # locale: US
    # ignore-words:
    #   - someword
  lll:
    # max line length, lines longer will be reported. Default is 120.
    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
    # line-length: 120
    # tab width in spaces. Default to 1.
    # tab-width: 1
  unused:
    # treat code as a program (not a library) and report unused exported identifiers; default is false.
    # XXX: if you enable this setting, unused will report a lot of false-positives in text editors:
    # if it's called for subdir of a project it can't find funcs usages. All text editor integrations
    # with golangci-lint call it on a directory with the changed file.
    # check-exported: false
  unparam:
    # Inspect exported functions, default is false. Set to true if no external program/library imports your code.
    # XXX: if you enable this setting, unparam will report a lot of false-positives in text editors:
    # if it's called for subdir of a project it can't find external interfaces. All text editor integrations
    # with golangci-lint call it on a directory with the changed file.
    # check-exported: false
  nakedret:
    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
    # max-func-lines: 30
  prealloc:
    # XXX: we don't recommend using this linter before doing performance profiling.
    # For most programs usage of prealloc will be a premature optimization.

    # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
    # True by default.
    # simple: true
    # range-loops: true # Report preallocation suggestions on range loops, true by default
    # for-loops: false # Report preallocation suggestions on for loops, false by default
  gocritic:
    # Which checks should be enabled; can't be combined with 'disabled-checks';
    # See https://go-critic.github.io/overview#checks-overview
    # To check which checks are enabled run `GL_DEBUG=gocritic golangci-lint run`
    # By default list of stable checks is used.
    # enabled-checks:
    #   - rangeValCopy

    # Which checks should be disabled; can't be combined with 'enabled-checks'; default is empty
    disabled-checks:
      - hugeParam # premature optimization
      - paramTypeCombine # questionable

    # Enable multiple checks by tags, run `GL_DEBUG=gocritic golangci-lint` run to see all tags and checks.
    # Empty list by default. See https://github.com/go-critic/go-critic#usage -> section "Tags".
    enabled-tags:
      - diagnostic
      - style
      - performance
      - experimental
      - opinionated

    settings: # settings passed to gocritic
      captLocal: # must be valid enabled check name
        paramsOnly: true
      rangeValCopy:
        sizeThreshold: 32

linters:
  # enable:
  #   - megacheck
  #   - govet
  enable-all: true
  disable:
    # - dupl
    # - gochecknoglobals
    # - gochecknoinits
    - lll
    # - misspell
    - prealloc
  # disable-all: false
  # presets:
  #   - bugs
  #   - unused
  fast: false


issues:
  # List of regexps of issue texts to exclude, empty list by default.
  # But independently from this option we use default exclude patterns,
  # it can be disabled by `exclude-use-default: false`. To list all
  # excluded by default patterns execute `golangci-lint run --help`
  exclude:
    - "declaration of \"(log|err|ctx)\" shadows"
    - "Potential file inclusion via variable" # [DEFAULT] gosec: False positive is triggered by 'src, err := ioutil.ReadFile(filename)'

  # Excluding configuration per-path, per-linter, per-text and per-source
  exclude-rules:
    # Exclude some linters from running on tests files.
    - path: _test\.go|testing\.go
      linters:
        - gocyclo
        - bodyclose
        - errcheck
        - dupl
        - gosec
        - gochecknoinits
        - gochecknoglobals

    # Ease some gocritic warnings on test files.
    - path: _test\.go|testing\.go
      text: "(unnamedResult|exitAfterDefer|rangeValCopy)"
      linters:
        - gocritic

    # Package def is designed to contain global constants.
    - path: internal/def/
      linters:
        - gochecknoglobals

    # Exclude known linters from partially hard-vendored code,
    # which is impossible to exclude via "nolint" comments.
    # - path: internal/hmac/
    #   text: "weak cryptographic primitive"
    #   linters:
    #     - gosec

    # Exclude some staticcheck messages
    # - linters:
    #     - staticcheck
    #   text: "SA9003:"

    # Exclude lll issues for long lines with go:generate
    # - linters:
    #     - lll
    #   source: "^//go:generate "

  # Independently from option `exclude` we use default exclude patterns,
  # it can be disabled by this option. To list all
  # excluded by default patterns execute `golangci-lint run --help`.
  # Default value for this option is true.
  exclude-use-default: false

  # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
  max-issues-per-linter: 0

  # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
  max-same-issues: 0

  # Show only new issues: if there are unstaged changes or untracked files,
  # only those changes are analyzed, else only changes in HEAD~ are analyzed.
  # It's a super-useful option for integration of golangci-lint into existing
  # large codebase. It's not practical to fix all existing issues at the moment
  # of integration: much better don't allow issues in new code.
  # Default is false.
  # new: false

  # Show only new issues created after git revision `REV`
  # new-from-rev: REV

  # Show only new issues created in git patch with set file path.
  # new-from-patch: path/to/patch/file
```

**Comment by abscondment** [comment]
I'm hitting this same issue with `v1.17.1` / `go1.12.5 linux/amd64`.

**Comment by ldez** [comment]
I created a patch but I'm not sure if it covers your case, could you try to isolate the code that produces the panic?

**Comment by powerman** [comment]
No, sorry, I've seen it just once, no idea how to reproduce.

**Comment by ldez** [comment]
You have to find the "body-close" pattern in your code base and try to change your code to find the lines that produce the panic when they are analyzed by the linter.

**Comment by aaronhnatiw** [comment]
I'm getting this same nil pointer dereference.  My stack trace is below.

I've also noted that #622 is a duplicate of this, and I've added a comment there to try to point everyone here.

OSX version 10.14.5
go version go1.12.7 darwin/amd64
golangci-lint has version 1.17.1 built from 4ba2155 on 2019-06-10T09:07:35Z

```shell
/usr/local/bin/golangci-lint run -c .golangci.yml ./...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1598484]

goroutine 1835 [running]:
go/types.(*object).Name(...)
	/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/go/types/object.go:134
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).isCloseCall(0xc00009e2c0, 0x1be0ea0, 0xc00798b700, 0xc0068132c0)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:230 +0x234
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).isopen(0xc00009e2c0, 0xc014b1f760, 0x9, 0xc01169dd00)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:175 +0x2e9
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).run(0xc00009e2c0, 0xc0000cbd60, 0x10, 0x191fec0, 0x4af2d33feb6c8001, 0xc012290720)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:100 +0x641
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).execOnce(0xc00041d720)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:382 +0x68a
sync.(*Once).Do(0xc00041d720, 0xc000a83790)
	/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/sync/once.go:44 +0xb3
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).exec(0xc00041d720)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:303 +0x50
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll.func1(0xc00041d720)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:291 +0x34
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:297 +0x11b

Process finished with exit code 2
```

**Comment by aaronhnatiw** [comment]
> You have to find the "body-close" pattern in your code base and try to change your code to find the lines that produce the panic when they are analyzed by the linter.

I believe it is happening in my code when I close a channel inside a goroutine, when that channel is created outside of the goroutine, but _not_ passed in as a parameter. Here's a snipped version of my code (the `close(done)` function happens in an `if` statement inside the goroutine, but I don't think that's relevant here):

```go
done := make(chan bool)
go func() {
    close(done)
}()
```

I did not get this error until I removed the channel reference from the goroutine's parameters (i.e. I _had_ been passing it in as a parameter to the goroutine before).

It may also be relevant, but I have multiple goroutines that all close the `done` channel from the same parent function.

**Comment by aaronhnatiw** [comment]
I've found the cause of the problem.

It crashed because I hadn't closed a response body from the server, which I had copied out as an `ioutil.NopCloser` type.

### Details

I had first read the body using the following function (taken directly from the go standard library):

```go
// ReadBody safely reads the HTTP request or response body, returning a byte
// slice of the body contents, a copy of the body (to add back into the body),
// and an error, if any.
func ReadBody(body io.ReadCloser) (contents []byte, copy io.ReadCloser, err error) {
	if body == http.NoBody {
		// No copying needed. Preserve the magic sentinel meaning of NoBody.
		return nil, http.NoBody, nil
	}
	var buf bytes.Buffer
	if _, err = buf.ReadFrom(body); err != nil {
		return nil, body, err
	}
	if err = body.Close(); err != nil {
		return nil, body, err
	}
	return buf.Bytes(), ioutil.NopCloser(bytes.NewReader(buf.Bytes())), nil
}
```

I then used it like so, which turns the body into a `ioutil.NopCloser` type:

```go
var respBody []byte
respBody, resp.Body, err = network.ReadBody(resp.Body)
```

So, the problem might be that it hits a `nil pointer dereference` error when it's trying to dereference an `ioutil.NopCloser` type, instead of the standard `io.ReadCloser` type that `http.Request.Body` and `http.Response.Body` use.

**Comment by palsivertsen** [comment]
Possibly fixed by https://github.com/timakin/bodyclose/pull/17

I was able to reproduce the issue with this code
```go
package main

import (
	"io"
	"io/ioutil"
	"log"
	"net/http"
)

var c http.Client

func main() {
	req, _ := http.NewRequest(http.MethodGet, "/", nil)
	resp, _ := c.Do(req)
	readAndClose(resp.Body)
}

// readAndClose read the body to EOF and closes it
func readAndClose(b io.ReadCloser) {
	if b == nil {
		return
	}

	if _, err := io.Copy(ioutil.Discard, b); err != nil {
		log.Printf("read response body: %v", err)
	}

	if err := b.Close(); err != nil {
		log.Printf("close response body: %v", err)
	}
}
```
And then running `golangci-lint run --disable-all --enable bodyclose`.

### System info
go version go1.12.6 linux/amd64
golangci-lint has version 1.17.1 built from 4ba2155 on 2019-06-10T09:06:49Z

**Comment by bwplotka** [comment]
Can still repro this on d2b1eea2c6171a1a1141a448a745335ce2e928a1 revision (latest) ):

```
>>  linting all of the Go files
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x99bda3]

goroutine 7795 [running]:
go/types.(*object).Name(...)
	/usr/local/go/src/go/types/object.go:134
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).isCloseCall(0xc04ed4ece0, 0xfe72a0, 0xc05fc8a5a0, 0xc01c418c00)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:228 +0x2a3
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.(*runner).isopen(0xc04ed4ece0, 0xc02a660e70, 0x2, 0xc02a198700)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:177 +0x2e9
github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose.runner.run(0xc04a18fcc0, 0xfefd20, 0xc015775f90, 0xc02e2e6b80, 0xfefdc0, 0xc00ed0b6d0, 0xc0061261e0, 0xc060c42fc0, 0xc04a18fcc0, 0x7f00124d5648, ...)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:102 +0x5a6
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).execOnce(0xc001e266e0)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:382 +0x68a
sync.(*Once).Do(0xc001e266e0, 0xc001b2af90)
	/usr/local/go/src/sync/once.go:44 +0xb3
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.(*action).exec(0xc001e266e0)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:303 +0x50
github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll.func1(0xc001e266e0)
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:291 +0x34
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker.execAll
	/tmp/thanos-go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/checker/checker.go:297 +0x11b
Makefile:237: recipe for target 'lint' failed
```

**Comment by bwplotka** [comment]
`golangci-lint run --print-resources-usage --enable goimports --enable misspell --enable goconst --enable bodyclose ./...`

**Comment by rittneje** [comment]
We are getting sporadic failures as well. I think you just need to pull https://github.com/timakin/bodyclose/pull/17 into your forked version.

**Comment by symbiont-james-sedgwick** [comment]
Any chance of cutting a new release soon? We're hitting this fairly frequently.

**Comment by stevenh** [comment]
This is also blocking us updating, which is a problem as we've moved to 1.13 so are hitting the go mod issues.

**Comment by jirfag** [comment]
@bwplotka It's another bug in bodyclose - https://github.com/timakin/bodyclose/pull/20.

**Comment by stevenh** [comment]
Unfortunately this doesn't fix all the issues, with 1.19.0 we still get a panic:
```
WARN [linters context] Panic: bodyclose: package "phttp" (isInitialPkg: true, needAnalyzeSource: true): runtime error: invalid memory address or nil pointer dereference: goroutine 47307 [running]:
runtime/debug.Stack(0xf57075, 0x3c, 0xc00167d8b8)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1(0xc034cff740)
        /Users/denis/go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:404 +0x1af
panic(0xd9a0e0, 0x1720290)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
go/types.(*object).Name(...)
        /usr/local/go/src/go/types/object.go:133
github.com/timakin/bodyclose/passes/bodyclose.(*runner).calledInFunc(0xc00167dc78, 0xc0b624d7c0, 0xc0b6512800, 0x10ae3a0)
        /Users/denis/go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:328 +0x284
github.com/timakin/bodyclose/passes/bodyclose.(*runner).isopen(0xc00167dc78, 0xc0b6476580, 0x14, 0x0)
        /Users/denis/go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:148 +0x555
github.com/timakin/bodyclose/passes/bodyclose.runner.run(0xc0a41bfc20, 0x10b9cc0, 0xc020a70910, 0xc0b6fc58e0, 0x10b9d60, 0xc020aea690, 0xc003d10c80, 0xc0b6f71a10, 0xc0a41bfc20, 0xe41140933e432942, ...)
        /Users/denis/go/src/github.com/golangci/golangci-lint/vendor/github.com/timakin/bodyclose/passes/bodyclose/bodyclose.go:102 +0x589
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0xc034cff740)
        /Users/denis/go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:495 +0x87a
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0xc034cff740)
        /Users/denis/go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:407 +0x5b
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func3(0xc09f24a800, 0xc034cff740)
        /Users/denis/go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:961 +0x69
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
        /Users/denis/go/src/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:956 +0x319 
````

**Comment by stevenh** [comment]
Raised a new issue to tack this here:
https://github.com/golangci/golangci-lint/issues/733
